tests/test_frame.py

```diff
@@ -2256,6 +2256,21 @@ def test_frame_stack() -> None:
             ),
             pd.Series,
         )
+        check(
+            assert_type(
+                df_multi_level_cols2.stack(0, dropna=True, sort=True),
+                Union[pd.DataFrame, "pd.Series[Any]"],
+            ),
+            pd.DataFrame,
+        )
```

Comment:
Looks like you have to include the `if PD_LTE_22` on this test as well, as the `dropna` argument will also change behavior in 3.0, so have one `if` statement surround both of the new tests you added.

To see this, do `poe pytest --nightly` on your current branch, and when you make the fix the problem will go away.

Reply:
Oh ok nice I didn't notice the `--nightly` option to test with pandas 3! It seems to pass now